adlfs/spec.py

```diff
@@ -1769,20 +1772,23 @@ def connect_client(self):
                 f"Unable to fetch container_client with provided params for {e}!!"
             )
 
-    async def _async_fetch_range(self, start: int, end: int, **kwargs):
+    async def _async_fetch_range(self, start: int, length: int = None, **kwargs):
         """
-        Download a chunk of data specified by start and end
+        Download a chunk of data specified by start and length
 
         Parameters
         ----------
         start: int
             Start byte position to download blob from
-        end: int
-            End byte position to download blob from
+        length: int
+            Length to download
         """
+        if length is not None:
+            if start + length > self.size:
+                length = self.size - start
         async with self.container_client:
             stream = await self.container_client.download_blob(
-                blob=self.blob, offset=start, length=end
+                blob=self.blob, offset=start, length=length
```

Comment:
...and then this line instead simply becomes
```suggestion
                blob=self.blob, offset=start, length=end-start
```

Reply:
Thanks @anders-kiaer.  I appreciate the feedback, and the confirmation that this issue resolves the challenge cited in #57.

I'll align the params in fetch_range to fsspec.AbstractBufferedFile, but I also want to account for the situation where end > self.size, and also for the eventuality that length is None, which is valid for the Azure SDK.  Can you take a look at this branch and provide feedback on performance?

Reply:
Good point on the `end=None` case (and transforming that into `length=None` - that should be handled if `None` is valid input in the `fsspec` `_fetch_range` function.
> Can you take a look at this branch and provide feedback on performance?


 I'll add some quick benchmarks I did based on this branch.